src/components/IFrame.js

```diff
@@ -50,16 +53,57 @@ export default class IFrame extends React.Component<Props, State> {
     window.addEventListener('message', this.handleMessage);
   }
 
-  componentWillUnmount(){
+  componentWillUnmount() {
     window.removeEventListener('message', this.handleMessage);
   }
 
+  buildSrc = (): string => {
+    const {checkoutId, chargeId, customMetadata} = this.props;
+
+    function encodeURIParams(params) {
+      let encoded = [];
+      let quote = window.encodeURIComponent;
+      for (let key in params) {
+        if (params.hasOwnProperty(key)) {
+          encoded.push(quote(key) + '=' + quote(params[key]));
+        }
+      }
+      return encoded.join('&');
+    }
+
+    let widgetType;
+    let id;
+    if (checkoutId) {
+      id = checkoutId;
+      widgetType = 'checkout';
+    } else if (chargeId) {
+      id = chargeId;
+      widgetType = 'charges';
+    } else {
+      throw new Error('must supply either checkoutId or chargeId prop');
+    }
+
+    let custom = '';
+    if (customMetadata && typeof customMetadata !== 'string') {
+      console.error('Received customMetadata not of "string" type. Ignoring.');
+    } else if (customMetadata) {
+      custom = customMetadata
+    }
+    const params = {
+      origin: this.hostName,
+      version: VERSION,
+      buttonId: this.uuid,
+      custom
+    };
```

Comment:
it doesn't really matter, but i would find it cleaner to not send the `custom=` query param when it's missing